src/frontend/screens/Audio/AudioRecording/index.tsx

```diff
@@ -0,0 +1,138 @@
+
+const m = defineMessages({
+  lessThan5: {
+    id: 'screens.AudioRecording.lessThan5',
+    defaultMessage: 'Less than 5 minutes left',
+  },
+  lessThan1: {
+    id: 'screens.AudioRecording.lessThan1',
+    defaultMessage: 'Less than a minute left',
+  },
+  record5Min: {
+    id: 'screens.AudioRecording.record5Min',
+    defaultMessage: 'Record up to 5 minutes',
+  },
+});
+
+export function AudioRecording({
+  navigation,
+}: NativeRootNavigationProps<'AudioRecording'>) {
+  const {startRecording, stopRecording, status, error, setError, reset} =
+    useAudioRecording();
+
+  const timeElapsed = status?.durationMillis || 0;
+  const isRecording = !!status?.isRecording;
+
+  const {formatMessage} = useIntl();
+
+  usePreventBackButtonWhileRecording({
+    shouldPrevent: isRecording,
+  });
+
+  const finishRecording = React.useCallback(() => {
+    stopRecording()
+      .then(uri => {
+        if (!uri) {
+          throw new Error('Recording is done, but no URI is available.');
+        }
+        navigation.replace('AudioPlaybackUnsavedReview', {
+          uri,
+          duration: timeElapsed,
+        });
+      })
+      .catch(err => {
+        setError(err);
+      });
+  }, [stopRecording, setError, navigation, timeElapsed]);
+
+  // stop recording at 5 minutes
+  React.useEffect(() => {
+    if (timeElapsed >= MAX_RECORDING_DURATION_MS) {
+      finishRecording();
+    }
+  }, [timeElapsed, finishRecording]);
+
+  return (
+    <>
+      <ScreenContentWithDock
+        contentContainerStyle={AudioStyles.contentContainer}
+        dockContainerStyle={AudioStyles.dockContainer}
+        dockContent={
+          status?.isDoneRecording ? (
+            <View
+              style={{
+                justifyContent: 'center',
+                alignItems: 'center',
+                paddingBottom: 60,
+                marginTop: 20,
+              }}>
+              <UIActivityIndicator
+                color={WHITE}
+                size={PRIMARY_CONTROL_DIAMETER}
+              />
+            </View>
+          ) : (
+            <TouchableOpacity
+              onPress={!isRecording ? startRecording : finishRecording}
+              style={AudioStyles.basePressable}>
+              {<View style={!isRecording ? styles.record : styles.stop} />}
+            </TouchableOpacity>
+          )
+        }>
+        <View style={AudioStyles.container}>
+          <View style={AudioStyles.timerContainer}>
+            <Text
+              style={[
+                AudioStyles.timerText,
+                {color: !isRecording ? MEDIUM_GREY : WHITE},
+              ]}>
+              {Duration.fromMillis(timeElapsed).toFormat('mm:ss')}
+            </Text>
+          </View>
+          <HeaderText variant="header3" style={AudioStyles.message}>
+            {!isRecording
+              ? formatMessage(m.record5Min)
+              : timeElapsed < 240000
+                ? formatMessage(m.lessThan5)
+                : formatMessage(m.lessThan1)}
```

Comment:
aside from flipping the ternary and ideally avoiding using a nested ternary, I would primarily suggest restoring the usage of the [plural syntax](https://formatjs.github.io/docs/core-concepts/icu-syntax#plural-format) for this.

1. Replace `m.lessThan5` and `m.lessThan1` with:

```ts
  minutesRemaining: {
    id: 'screens.AudioRecording.minutesRemaining',
    defaultMessage:
      'Less than {value, plural, one {a minute} other {{value, number} minutes}} left',
  },
```

2. Use it here

```tsx
            {isRecording
              ? formatMessage(m.minutesRemaining, {
                  value: timeElapsed < 240_000 ? 5 : 1,
                })
              : formatMessage(m.record5Min)}
```

Reply:
This feels like this adds unnecessary complexity without a clear maintainability benefit. Given our agreed-upon principles of keeping things maintainable and avoiding premature optimizations, I’d suggest keeping the existing approach, which is simpler and easier to understand at a glance.

Reply:
I agree it's a bit harder to understand at a glance. Not opposed to keeping what you have, but I wouldn't be surprised if the suggested approach is a generally better experience for translators. Can't really confirm this but I would imagine that keeping these as the same message allows the translator to understand that this message is used for the same UI, whereas having two messages doesn't guarantee that. Purely conjecture though!